pkg/controller/incompatible_operator_controller.go

```diff
@@ -0,0 +1,181 @@
+	defer c.logger.Info("sync finished")
+
+	var updateStatusFn v1helpers.UpdateStatusFunc
+	incompatibleOperators, err := c.getIncompatibleOperators()
+	if len(incompatibleOperators) > 0 {
+		updateStatusFn = v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    typeIncompatibelOperatorsUpgradeable,
+			Status:  operatorv1.ConditionFalse,
+			Reason:  reasonIncompatibleOperatorsInstalled,
+			Message: strings.Join(incompatibleOperators, ","),
+		})
+	} else {
+		if err != nil {
+			updateStatusFn = v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Type:    typeIncompatibelOperatorsUpgradeable,
+				Status:  operatorv1.ConditionFalse,
+				Reason:  reasonInvalidBundleProperties,
+				Message: err.Error(),
+			})
+		} else {
+			updateStatusFn = v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Type:   typeIncompatibelOperatorsUpgradeable,
+				Status: operatorv1.ConditionTrue,
+			})
+		}
+	}
+
+	if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, updateStatusFn); updateErr != nil {
+		c.logger.V(4).Info(fmt.Sprintf("Error updating operator condition status: %v", updateErr))
+		return updateErr
+	}
+	return err
+}
+
+func (c *incompatibleOperatorController) getIncompatibleOperators() ([]string, error) {
+	var incompatibleOperators []string
+
+	ceList, err := c.clusterExtensionClient.Informer().Lister().List(labels.NewSelector())
+	if err != nil {
+		c.logger.V(4).Error(err, "Error listing cluster extensions")
+		return nil, err
+	}
+
+	store := c.buildHelmStore(c.kubeclient.CoreV1().Secrets("openshift-operator-controller"))
+
+	var errs []error
+	// Get all ClusterExtensions incompatible with next Y-stream
+	for _, obj := range ceList {
+		metaObj, ok := obj.(metav1.Object)
+		if !ok {
+			errs = append(errs, fmt.Errorf("metav1.Object type assertion failed for object %v", obj))
+			continue
+		}
+		name := metaObj.GetName()
+		rel, err := store.Deployed(name)
+		if errors.Is(err, driver.ErrNoDeployedReleases) {
+			c.logger.Info("Cluster Extension not yet deployed - will check again later")
+			continue
+		}
+		if err != nil {
+			c.logger.V(4).Error(err, "Error returning the last deployed release")
+			return nil, err
+		}
+
+		if rel.Chart == nil || rel.Chart.Metadata == nil {
+			continue
+		}
+		props, err := propertyListFromPropertiesAnnotation(rel.Chart.Metadata.Annotations["olm.properties"])
+		if err != nil {
+			c.logger.V(4).Error(err, "Error converting olm.properties")
+			return nil, err
+		}
+		numMaxOCPProps := 0
+		for _, p := range props {
+			if p.Type == maxOpenShiftVersionProperty {
+				numMaxOCPProps++
+				maxOCPVersion, err := utils.ToAllowedSemver(string(p.Value))
+				if err != nil {
+					errs = append(errs, fmt.Errorf(fmt.Sprintf("Error converting to semver for version %s: %v", string(p.Value), err)))
+					continue
+				}
+				if numMaxOCPProps > 1 {
+					errs = append(errs, fmt.Errorf("%s has more than one %s", name, maxOpenShiftVersionProperty))
```

Comment:
Same here about including ClusterExtension and bundle name context in the error.